platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -265,6 +268,19 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
         } else {
             registration.addProvider(VintageModelProvider())
         }
+
+        if (allowNudgingToEnableConfigurationCache(requirements, modelParameters)) {
+            // Don't suggest enabling CC if it is on, even if implicitly (e.g. enabled by isolated projects).
+            // Most likely, the user who tries IP is already aware of CC and nudging will be just noise.
+            registration.add(ConfigurationCachePromoHandler::class.java)
+        } else if (!modelParameters.isConfigurationCache) {
```

Comment:
🤔 Will any `ProblemListener` instance be added in the DI graph in case of mismatch this `else if` condition?

`DefaultConfigurationCacheProblemsListener` requires `ProblemListener` service be injected. 
- For CC we have `ConfigurationCacheProblems `
- For nudging we have `ConfigurationCachePromoHandler `
- For noCC we have `IgnoringProblemsListener`

Having written this, I see exhaustiveness in `CC + noCC` branches. Initially I thought it would be a little more safe to have just `else` in this line to have some `ProblemListener` implementation injected anyways

Reply:
> Having written this, I see exhaustiveness in `CC + noCC` branches. Initially I thought it would be a little more safe to have just `else` in this line to have some `ProblemListener` implementation injected anyways

Thanks, I've adjusted the registration code to select one of three in a more explicit fashion.